### ISSUE TITLE
Problems fqdn <> cername, and other configuration issues 

### ISSUE THREAD

**Issue by t4nec0** [tedivm/puppet-rsnapshot#5]
Hi,
compliments for this great module. I'm trying to use it in puppet 4.2 + PuppetDB 3.0.2. I've configured the server and the client successfully but I'm stuck at the point that the client configuration isn't replicating onto the server.
The module documentation say that storeconfig is required, and I have this in `puppet.conf`:

> [master]
> ...
> storeconfigs = true
> storeconfigs_backend = puppetdb

However, it doesn't seem to work as expected.
If you have any advice about where I'm doing it wrong...
Thanks again.

  Ivan

**Comment by t4nec0** [comment]
Tried also to upgrade puppetdb to the latest 3.2, nothing apparently changed.

In the client's log I can read:

> puppet-agent[12442]: Public key from root@$BACKUPSERVER (for local user backshots) not available yet. Skipping
> puppet-agent[12442]: (/Stage[main]/Rsnapshot::Client::User/Sshkeys::Set_authorized_key[root@$BACKUPSERVER to backshots]/Notify[Public key from root@$BACKUPSERVER (for local user backshots) not available yet. Skipping]/message) defined 'message' as 'Public key from root@$BACKUPSERVER (for local user backshots) not available yet. Skipping'

**Comment by t4nec0** [comment]
I think I solved the sshkeys problems. The issue was that the fqdn of the server was different than its certname. jtopjian-sshkeys relies on fqdn so the key was not found in puppetdb.
The fix consists in putting the wanted fqdn as first entry in /etc/hosts of the server, facter will use that.

**Comment by t4nec0** [comment]
OK, that seems to have solved the problem. However, I stumbled upon other issues:
- all the undef settings in params.pp do not work well in puppet 4 when using the template for the client config (it's a problem related to string comparison in the template). I quickly solved by changing to 'undef' in the params.pp file, but it's far from ideal.
- the first rsync stops asking to confirm the ssh fingerprint. Had to add 

> Host *
>     StrictHostKeyChecking no

into the servers' `/etc/ssh/ssh_config` file. I tried adding `-oStrictHostKeyChecking=no` in the `ssh_params`, again in params.pp, but with no results.
